lib/ecto/embedded.ex

```diff
@@ -30,7 +31,32 @@ defmodule Ecto.Embedded do
       cardinality: Keyword.fetch!(opts, :cardinality),
       owner: module,
       embed: Keyword.fetch!(opts, :embed),
-      container: Keyword.get(opts, :container)
+      container: Keyword.get(opts, :container),
+      changeset: Keyword.fetch!(opts, :changeset)
     }
   end
+
+  @doc """
+  Normalizes the changeset function
+
+  The new_fun can have formats as the changeset field. Returns the struct
+  with the changeset field updated to have a changeset fun, that can be
+  called with params.
+  """
+  def normalize_changeset(%__MODULE__{embed: module} = embed, new_fun) do
+    update_in embed.changeset, &do_normalize_changeset(&1, module, new_fun)
+  end
+
+  defp do_normalize_changeset({mod, fun}, _module, nil),
```

Comment:
We should not keep anonymous functions around because they are harder to serialize. If you want to normalize, normalize it to `{module, function}` at compile time. Also, we could make this function private and call it when we build the embedding, instead of it being public and calling it later.